src/modules/avformat/factory.c

```diff
@@ -437,3 +439,251 @@ MLT_REPOSITORY
+	av_opt_set_int( sws, "dst_format", ctx->dst_frame->format, 0 );
+	av_opt_set_int( sws, "sws_flags", ctx->flags, 0 );
+
+	av_opt_set_int( sws, "src_h_chr_pos", -513, 0 );
+	av_opt_set_int( sws, "src_v_chr_pos", src_v_chr_pos, 0 );
+	av_opt_set_int( sws, "dst_h_chr_pos", -513, 0 );
+	av_opt_set_int( sws, "dst_v_chr_pos", dst_v_chr_pos, 0 );
+
+	if ( ( ret = sws_init_context( sws, NULL, NULL ) ) < 0 )
+	{
+		mlt_log_error( NULL, "%s:%d: sws_init_context failed, ret=%d\n", __FUNCTION__, __LINE__, ret );
+		sws_freeContext( sws );
+		return 0;
+	}
+
+	avformat_set_luma_transfer( sws, ctx->src_colorspace, ctx->dst_colorspace, ctx->src_full_range, ctx->dst_full_range );
+
+#if LIBAVUTIL_VERSION_INT < AV_VERSION_INT(55, 0, 100)
+#define PIX_DESC_BPP(DESC) (DESC.step_minus1 + 1)
+#else
+#define PIX_DESC_BPP(DESC) (DESC.step)
+#endif
+
+#if LIBAVUTIL_VERSION_INT < AV_VERSION_INT(52, 32, 100)
+#define AV_PIX_FMT_FLAG_PLANAR PIX_FMT_PLANAR
+#endif
+	for( i = 0; i < 4; i++ )
+	{
+		int in_offset = (AV_PIX_FMT_FLAG_PLANAR & ctx->src_desc->flags)
+			 ? ( ( 1 == i || 2 == i ) ? ( slice_x >> ctx->src_desc->log2_chroma_w ) : slice_x )
+			 : ( ( i ) ? 0 : slice_x * PIX_DESC_BPP(ctx->src_desc->comp[0]) );
+
+		int out_offset = (AV_PIX_FMT_FLAG_PLANAR & ctx->dst_desc->flags)
+			 ? ( ( 1 == i || 2 == i ) ? ( slice_x >> ctx->dst_desc->log2_chroma_w ) : slice_x )
+			 : ( ( i ) ? 0 : slice_x * PIX_DESC_BPP(ctx->dst_desc->comp[0]) );
+
+		in_stride[i]  = ctx->src_frame->linesize[i] * mul;
+		out_stride[i] = ctx->dst_frame->linesize[i] * mul;
+
+		in[i] =  ctx->src_frame->data[i] + ctx->src_frame->linesize[i] * field + in_offset;
+		out[i] = ctx->dst_frame->data[i] + ctx->dst_frame->linesize[i] * field + out_offset;
+	}
+
+	sws_scale( sws, in, in_stride, 0, h, out, out_stride );
+
+	sws_freeContext( sws );
+
+	return 0;
+}
+
+int avformat_colorspace_convert
+(
+	AVFrame* src, int src_colorspace, int src_full_range,
+	AVFrame* dst, int dst_colorspace, int dst_full_range,
+	int flags
+)
+{
+	int c, i;
+
+	struct sliced_pix_fmt_conv_t ctx =
+	{
+		.flags = flags,
+		.src_frame = src,
+		.dst_frame = dst,
+		.src_colorspace = src_colorspace,
+		.dst_colorspace = dst_colorspace,
+		.src_full_range = src_full_range,
+		.dst_full_range = dst_full_range,
+	};
+
+	ctx.src_desc = av_pix_fmt_desc_get( ctx.src_frame->format );
+	ctx.dst_desc = av_pix_fmt_desc_get( ctx.dst_frame->format );
+
+	if ( !getenv("MLT_AVFORMAT_SLICED_PIXFMT_DISABLE") )
+		ctx.slice_w = ( src->width < 1000 )
+			? ( 256 >> src->interlaced_frame )
+			: ( 512 >> src->interlaced_frame );
+	else
+		ctx.slice_w = src->width;
+
+	c = ( src->width + ctx.slice_w - 1 ) / ctx.slice_w;
+	c *= src->interlaced_frame ? 2 : 1;
+
+	if ( !getenv("MLT_AVFORMAT_SLICED_PIXFMT_DISABLE") )
+		mlt_slices_run_normal( c, sliced_h_pix_fmt_conv_proc, &ctx );
+	else
+		for ( i = 0 ; i < c; i++ )
+			sliced_h_pix_fmt_conv_proc( i, i, c, &ctx );
+
+	return dst_colorspace;
+}
+#endif
+
+int avformat_map_pixfmt_av2mlt( int f )
```

Comment:
This function seems to serve same purpose as pick_image_format()  but it is different and not yet used. I think it should be expanded to map more of the av pixel formats to be a suitable replacement for pick_image_format(). Also, it should take enum AVPixelFormat as the argument and return mlt_image_format.